src/arch/helperrvv.h

```diff
@@ -742,20 +743,10 @@ static INLINE vint vcast_vi_i(int32_t i) {
   return SLEEF_RVV_DP_VCAST_VI_I(i, VECTLENDP);
 }
 static INLINE vint vrint_vi_vd(vdouble vd) {
-  // It is not currently possible to safely set frm for intrinsics,
-  // so emulate round-to-nearest behavior
-  vdouble half = SLEEF_RVV_DP_VCAST_VD_D(0.5, VECTLENDP);
-  half = __riscv_vfsgnj(half, vd, VECTLENDP);
-  vdouble res = __riscv_vfadd(vd, half, VECTLENDP);
-  return __riscv_vfncvt_rtz_x(res, VECTLENDP);
+  return SLEEF_RVV_DP_VFNCVT_X_F_VI(vd, __RISCV_VXRM_RNU, VECTLENDP);
 }
 static INLINE vdouble vrint_vd_vd(vdouble vd) {
-  // It is not currently possible to safely set frm for intrinsics,
-  // so emulate round-to-nearest behavior
-  vdouble half = SLEEF_RVV_DP_VCAST_VD_D(0.5, VECTLENDP);
-  half = __riscv_vfsgnj(half, vd, VECTLENDP);
-  vdouble res = __riscv_vfadd(vd, half, VECTLENDP);
-  return __riscv_vfwcvt_f(__riscv_vfncvt_rtz_x(res, VECTLENDP), VECTLENDP);
+  return SLEEF_RVV_DP_VFCVT_F_X_VD(SLEEF_RVV_DP_VFCVT_X_F_VD_RM(vd, __RISCV_VXRM_RNU, VECTLENDP), VECTLENDP);
```

Comment:
I believe these should use `__RISCV_FRM_RNE` since [these instructions](https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#1317-single-width-floating-pointinteger-type-convert-instructions) are controlled by `frm`, not the fixed-point rounding mode, as reflected in [their intrinsics' API](https://github.com/riscv-non-isa/rvv-intrinsic-doc/blob/main/auto-generated/intrinsic_funcs/04_vector_floating-point_intrinsics.adoc#vector-floating-point-move-intrinsics).  Same with all the similar ones above.

Reply:
I tried with passing `__RISCV_FRM_RNE` here and it didn't work. Given on the non-vector floating-point rounding mode, RNE is 0, and on the vector one, RNU is 0, I just passed "0" aka RNU. Do you know of documentation for what RNU means? I couldn't find anything...

Reply:
It's quite interesting that you observe different behavior in the two cases, since both of those enum values should map to zero, and do when I tested them: https://godbolt.org/z/YhPWMeMsE.  I would be interested to see the preprocessed output with both values, and to know which toolchain you are using.

I agree that the fixed-point rounding mode semantics match my original code, but these instructions are not fixed-point and are not controlled by the `vxrm` register.

Reply:
Actually, I just realized you pointed to "FRM" while I was using "VXRM", sorry. Let me try that on Monday as well.

Reply:
No worries. Thanks for looking into this, and have a great weekend!